artiq/language/environment.py

```diff
@@ -242,6 +252,8 @@ def __init__(self, managers_or_parent, *args, **kwargs):
         self.__in_build = True
         self.build(*args, **kwargs)
         self.__in_build = False
+        if isinstance(self.__argument_mgr, ProcessArgumentManager):
+            self.__argument_mgr.check_unused_arguments()
```

Comment:
Is this the best place to check for this? Why?

Reply:
I think the best place to check is right after setting all arguments, which should be after build stage.